content/how-to-persist-app-data-across-os-updates.md

```diff
@@ -0,0 +1,58 @@
+---
+title: How to Persist Application Data across Torizon OS Updates
+date: 2024-12-02
+author: Leonardo Graboski Veiga, Product Owner Torizon OS
+draft: false
+tags:
+    - Container
+    - Embedded
+    - Linux
+abstract: How to make sure your application data is not overwritten or erased by OS Updates
+image: ./types-of-mounts.webp
+---
+
+The Torizon OS is designed around the concept that different parties handle _applications_ and _OS_ development:
+
+- Applications: you, as a Torizon customer, develop and update your application on your development cycle
+- OS: Toradex, more specifically the Torizon OS Team, develops and updates the base OS, releasing it every quarter
+
+A question I sometimes get from customers is: how can I keep my application data when updating the base OS?
+
+## A Practical Answer
+
+**You must store the application data somewhere under the `/var` directory**. OSTree, the file system management tool used for OS Updates, keeps the `/var` directory intact during the updates.
+
+Therefore, the two standard Docker [storage methods](https://docs.docker.com/engine/storage/) are ok to use.
+
+![Types of Mounts](/types-of-mounts.webp)
+*Figure 1: Types of Mounts (source: Docker Docs)*
+
+You can use [Bind Mounts](https://docs.docker.com/engine/storage/bind-mounts/), as long as keeping the mount somewhere under `/var`. The `/home/torizon` directory is recommended, as it is symlinked to `/var` and owned by the `torizon` user.
+
+Here is an example that shares a directory named `my-app` from the torizon home directory to a container that expects to store data in the directory named `my-app` under `/etc`:
+
+```sh
+docker run -v /home/torizon/my-app:/etc/my-app app-container
+```
+
+You can use [Volumes](https://docs.docker.com/engine/storage/volumes/), as they are effectively stored under /var by default, specifically under /var/lib/docker/volumes. Therefore you can safely use them as documented by Docker.
```

Comment:
Add a "Or, you can also use [Volumes..." so it reads better.

Reply:
Got it!